sound/soc/sof/intel/hda-dsp.c

```diff
@@ -442,7 +442,14 @@ static int hda_dsp_set_D0_state(struct snd_sof_dev *sdev,
 	/* fallback to the previous register value */
 	value = value ? 0 : SOF_HDA_VS_D0I3C_I3;
 
-	return hda_dsp_update_d0i3c_register(sdev, value);
+	/*
+	 * This can fail but return the IPC error to signal that
+	 * the state change failed. The state transition will not be retried
+	 * if it fails until the set_power_state() op gets called again.
```

Comment:
by a new IPC.

Reply:
hmm that would be too specific. The op could be called again because of a new IPC or even a system suspend call.

Reply:
then explain further. You've said too much or too little at this point

Reply:
hmm ok. I'added a bit more explanation for what happens when the state transition fails. Is this better?